web/client-api/src/main/java/io/deephaven/web/client/api/subscription/TableViewportSubscription.java

```diff
@@ -74,21 +83,40 @@ public class TableViewportSubscription extends AbstractTableSubscription {
      */
     private boolean retained;
 
-
     private UpdateEventData viewportData;
 
-    public TableViewportSubscription(double firstRow, double lastRow, Column[] columns, Double updateIntervalMs,
-            JsTable existingTable) {
-        super(existingTable.state(), existingTable.getConnection());
-        this.firstRow = firstRow;
-        this.lastRow = lastRow;
-        this.columns = columns;
+    public static TableViewportSubscription make(double firstRow, double lastRow, Column[] columns,
+            Double updateIntervalMs, JsTable existingTable) {
+        ClientTableState tableState = existingTable.state();
+        WorkerConnection connection = existingTable.getConnection();
+
+        final ClientTableState stateToSubscribe;
+        ConfigValue flattenViewport = connection.getServerConfigValue("web.flattenViewports");
```

Comment:
Guess we don't really have a good place to put this constant or have a convenience method to get it as a bool...